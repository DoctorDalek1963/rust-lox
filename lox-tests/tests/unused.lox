fun f(x) {
  var a = 1;
  var b = 2;
  var c;
  print a;

  fun g() {
    print "do nothing";
  }
}

f(nil);
