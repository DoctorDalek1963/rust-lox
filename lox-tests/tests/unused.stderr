WARNING: Local function 'g' is never used
 --> 7:3
  |
7 |   fun g() {
  |   ^-----^

WARNING: Parameter 'x' is never used
 --> 1:7
  |
1 | fun f(x) {
  |       ^

WARNING: Local variable 'b' is never used
 --> 3:3
  |
3 |   var b = 2;
  |   ^--------^

WARNING: Local variable 'c' is never used
 --> 4:3
  |
4 |   var c;
  |   ^----^
